### ISSUE TITLE
Feature Request: Support encryption

### ISSUE THREAD

**Issue by huntzhan** [hackjutsu/Lepton#363]
The more we know about your system and use case, the more easily and likely we can help.

#### Environment info
* OS: N/A
* App Version (`Command/Ctrl + ,`): N/A

#### Description of the problem / feature request / question:

AFAIK, the secret gist means *not searchable* instead of *not accessible* by the public. I think it would be great to support encryption on the client side to make it really a secret, as such user could have less concern to the content they post.

#### If possible, provide the log files
- Mac: `~/Library/Application Support/Lepton/logs`
- Windows: `C:\Users\<username>\AppData\Roaming\Lepton\logs`
- Linux: `~/.cached/Lepton/logs`
>The [debug mode](https://github.com/hackjutsu/Lepton/wiki) is recommended when producing the logs. 
>You can send your log file to cosmo.lepton@gmail.com if don't want to post it here.

**Comment by hackjutsu** [comment]
@huntzhan Hello, could you clarify "encryption on the client side"? Lepton doesn't store this data locally and use GitHub APIv3 for remote update.

**Comment by huntzhan** [comment]
Thanks for the reply.

For example, the user could add an AES credential(password) to the config file, and then Lepton could help encrypt the gist with the credential before uploading to Github. Since only the encrypted content got pushed to Github, anyone without the credential could not recover the original information. Afterward, Lepton could help decrypt the file with the same credential, which is only stored locally.

**Comment by hackjutsu** [comment]
This is an interesting idea, but it will become a headache when users switch to another gist clients.

**Comment by fugkco** [comment]
@hackjutsu this is true, however, this wouldn't Lepton's problem. As long as there's a way for a user decrypt the snippets, combine this with #264, and it should be easy for a user to migrate to another client. Outside that, the likelihood of a user using multiple clients (outside of github) is low.
